addons/payment_odoo/models/payment_acquirer.py

```diff
@@ -0,0 +1,45 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import hashlib
+import hmac
+
+from werkzeug import urls
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class PaymentAcquirer(models.Model):
+    _inherit = 'payment.acquirer'
+
+    provider = fields.Selection(
+        selection_add=[('odoo', "Odoo Payments")], ondelete={'odoo': 'set default'})
+    odoo_adyen_account_id = fields.Many2one(
+        related='company_id.adyen_account_id', required_if_provider='odoo')
```

Comment:
TODO check if required_if_provider works fine with related fields

Reply:
Also need to check if everything is fine if you empty the field from the company side (maybe need to set it to store=True to ensure the required_if_provider check is triggered in this case ?)

Reply:
`required_if_provider` does work with related fields.

About the second comment, there shouldn't be an issue with clearing the field from the company on the 'ORM' side (not really ORM but you see what I mean).
Functionally, this is still a relevant question tho. I wouldn't prevent the action since the acquirer *uses* the record but does not manage it. Yet, the acquirer cannot work without it. I guess that we could simply disable the acquirer if that happens?

@aprieels What do you think? In the end, I'll simply adapt my branch when rebasing on the changes that you'd make either in 14.0 or in master.